ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -157,6 +176,85 @@ export const ProfilePage = withUserProfile()(class extends React.Component<
       </div>;
     };
 
+    function renderInstitutionalAffiliationComponents(componentHandle: React.Component) {
+      if (requireInstitutionalVerification) {
+        const {verifiedInstitutionalAffiliation} = profile;
+        return verifiedInstitutionalAffiliation &&
+          renderVerifiedInstitutionalAffiliationComponents(verifiedInstitutionalAffiliation);
+      } else {
+        return renderOldInstitutionalAffiliationComponents(componentHandle);
+      }
+    }
+
+    function renderVerifiedInstitutionalAffiliationComponents(verifiedInstitutionalAffiliation: VerifiedInstitutionalAffiliation) {
+      const {institutionalRoleEnum, institutionalRoleOtherText} = verifiedInstitutionalAffiliation;
+      return verifiedInstitution && <React.Fragment>
+        <div style={{...styles.h1, marginBottom: 24}}>Verified Institutional Affiliation</div>
+        <FlexRow>
+          <FlexColumn>
+            <div style={styles.inputLabel}>Institution</div>
+            <div style={styles.uneditableProfileElement}>{verifiedInstitution.displayName}</div>
+          </FlexColumn>
+          <FlexColumn>
+            <div style={styles.inputLabel}>Role</div>
+            <div style={styles.uneditableProfileElement}>
+            {institutionalRoleEnum === InstitutionalRole.OTHER ? institutionalRoleOtherText : institutionalRoleEnum}
+            </div>
+          </FlexColumn>
+        </FlexRow>
+      </React.Fragment>;
+    }
+
+    function renderOldInstitutionalAffiliationComponents(componentHandle: React.Component) {
+      return <React.Fragment>
+        <div style={{...styles.h1, marginBottom: 24}}>Institution Affiliations</div>
```

Comment:
Moved from the existing `render()` with `this.setState()` changed to `componentHandle.setState()`